src/main/java/ch/hearc/todont/models/User.java

```diff
@@ -55,7 +45,78 @@
 
     // METHODS
 
-    public User() {}
+    /**
+     * Pledges this user to the given ToDont
+     * 
+     * @param toDont ToDont the user will be pledged to
+     * @return The new pledge object created in this method, or an existing pledge
+     *         to the Todont if it exists
+     */
+    public Pledge pledgeTo(ToDont toDont) {
+        Pledge pledge = getPledgeTo(toDont);
+
+        if (pledge == null) {
```

Comment:
Do we need to create a pledge here if it doesn't exists? Where would this be used?

Reply:
When a user clicks on "join todont" (or whatever the naming will be).

Actually, I'd rather remove the "return the pledge if it already exists" part of the method. Not sure about the semantics of it all.

The point of this method is to create a Pledge object joining the user to the ToDont in order to save it afterwards.

Reply:
My bad, I thought you were creating a ToDont. Totally makes sense now 👍